portal/views/practitioner.py

```diff
@@ -0,0 +1,311 @@
+
+
+@practitioner_api.route('/practitioner/<string:id_or_code>')
+@oauth.require_oauth()
+def practitioner_get(id_or_code):
+    """Access to the requested practitioner as a FHIR resource
+
+    If 'system' param is provided, looks up the user by identifier,
+    using the `id_or_code` string as the identifier value; otherwise,
+    treats `id_or_code` as the practitioner.id
+
+    ---
+    operationId: practitioner_get
+    tags:
+      - Practitioner
+    produces:
+      - application/json
+    parameters:
+      - name: id_or_code
+        in: path
+        description: TrueNTH practitioner ID OR Identifier value code
+        required: true
+        type: string
+      - name: system
+        in: query
+        description: Identifier system
+        required: false
+        type: string
+    responses:
+      200:
+        description:
+          Returns the requested practitioner as a FHIR [practitioner
+          resource](http://www.hl7.org/fhir/DSTU2/practitioner.html) in JSON.
+      401:
+        description:
+          if missing valid OAuth token or logged-in user lacks permission
+          to view requested patient
+
+    """
+    system = request.args.get('system')
+    if system:
+        practitioner = lookup_practitioner_by_external_id(system, id_or_code)
+    else:
+        check_int(id_or_code)
+        practitioner = Practitioner.query.get_or_404(id_or_code)
+    return jsonify(practitioner.as_fhir())
+
+
+@practitioner_api.route('/practitioner', methods=('POST',))
+@oauth.require_oauth()
+@roles_required([ROLE.ADMIN, ROLE.SERVICE])
+def practitioner_post():
+    """Add a new practitioner.  Updates should use PUT
+
+    Returns the JSON FHIR practitioner as known to the system after adding.
+
+    Submit JSON format [Practitioner
+    Resource](https://www.hl7.org/fhir/DSTU2/practitioner.html) to add an
+    practitioner.
+
+    ---
+    operationId: practitioner_post
+    tags:
+      - Practitioner
+    produces:
+      - application/json
+    parameters:
+      - in: body
+        name: body
+        schema:
+          id: FHIRPractitioner
+          required:
+            - resourceType
+          properties:
+            resourceType:
+              type: string
+              description: defines FHIR resource type, must be Practitioner
+    responses:
+      200:
+        description:
+          Returns created [FHIR practitioner
+          resource](http://www.hl7.org/fhir/DSTU2/practitioner.html) in JSON.
+      400:
+        description:
+          if practitioner FHIR JSON is not valid
+      401:
+        description:
+          if missing valid OAuth token or logged-in user lacks permission
+          to view requested patient
+
+    """
+    if (not request.json or 'resourceType' not in request.json or
+            request.json['resourceType'] != 'Practitioner'):
+        abort(400, "Requires FHIR resourceType of 'Practitioner'")
+    try:
+        practitioner = Practitioner.from_fhir(request.json)
```

Comment:
Won't this silently overwrite an existing one on a match?  I think we should force a PUT to update and raise if they POST on an existing, no?

Reply:
This shouldn't overwrite anything existing, since this POST uses `from_fhir()` - which creates a new Practitioner and then updates it using `update_from_fhir()` - in contrast to the PUT, which actively looks up the Practitioner (using either ID or external identifier) and then just updates it using `update_from_fhir()`.

We could get _duplicates_ from this POST, but that's just because there aren't any fields we can place uniqueness constrictions on.

Reply:
Per our slack convo, added checks around identifiers in PUT and POST endpoints